R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+      if (!selectedResponse) {
+        EMMCsummary$addFootnote(.mmMessageNotResponse)
+      } else{
+        EMMCsummary$addFootnote(.mmMessageResponse)
+      }
+    }
+
+
+    EMMCsummary$addRows(tempRow)
+
+  }
+}
+
+
+# specific Bayesian
+.mmReadDataB      <- function(dataset, options, type = "BLMM") {
+  if (!is.null(dataset)) {
+    return(dataset)
+  } else{
+    if (type == "LMM") {
+      return(
+        readDataSetToEnd(
+          columns.as.numeric = options$dependentVariable,
+          columns.as.factor  = c(options$fixedVariables, options$randomVariables)
+        )
+      )
+    } else if (type == "GLMM") {
+      if (options$dependentVariableAggregation == "") {
+        return(readDataSetToEnd(
+          columns = c(
+            options$dependentVariable,
+            options$fixedVariables,
+            options$randomVariables
+          )
+        ))
+      } else{
+        return(readDataSetToEnd(
+          columns = c(
+            options$dependentVariable,
+            options$fixedVariables,
+            options$randomVariables,
+            options$dependentVariableAggregation
+          )
+        ))
+      }
+    }
+  }
+}
+.mmFitModelB      <- function(jaspResults, dataset, options, type = "BLMM") {
+  # hopefully fixing the random errors
+  contr.bayes <<- stanova::contr.bayes
+  stan_glmer  <- rstanarm::stan_glmer
+  if (!is.null(jaspResults[["mmModel"]]))
+    return()
+
+  mmModel <- createJaspState()
+
+
+  if (type == "BLMM") {
+    dependencies <- .mmDependenciesBLMM
+  } else if (type == "BGLMM") {
+    dependencies <- .mmDependenciesBGLMM
+  }
+  mmModel$dependOn(dependencies)
+
+  modelFormula <- .mmModelFormula(options, dataset)
+
+  if (type == "BLMM") {
+    model <- tryCatch(stanova::stanova(
+      formula           = as.formula(modelFormula$modelFormula),
+      check_contrasts   = "contr.bayes",
+      data              = dataset,
+      chains            = options$chains,
+      iter              = options$iteration,
+      warmup            = options$warmup,
+      adapt_delta       = options$adapt_delta,
+      control           = list(maxTreedepth = options$max_treedepth),
+      seed              = .getSeedJASP(options),
+      model_fun         = "lmer"
+    ), error = function(e) e )
+
+  } else if (type == "BGLMM") {
+    # needs to be evaluated in the global environment
+    glmmLink      <<- options$link
+    if (options$family == "neg_binomial_2") {
+      glmmFamily <<- rstanarm::neg_binomial_2(link = glmmLink)
+    } else if (options$family == "betar") {
+      glmmFamily <<- mgcv::betar(link = glmmLink)
+    } else if (options$family != "binomial_agg"){
+      tempFamily <<- options$family
+      glmmFamily <<- eval(call(tempFamily, glmmLink))
+    }
+
+    # I wish there was a better way to do this
+    if (options$family == "binomial_agg") {
+      glmmWeight <<- dataset[, .v(options$dependentVariableAggregation)]
+
+      model <- tryCatch(stanova::stanova(
```

Comment:
shouldn't this and the one in the else statement be `try`?

Reply:
yes, probably all of them, I never get into using try, it's kinda distracting to have the error printed in red but the flow uninterrupted...

Reply:
There's `try(..., silent = TRUE)` if you don't want to see the error message. But `result <- try(...)` is practically identical to `result <- tryCatch(..., error = function(e))`, but much easier to read because there won't be multiple catch statements.